### PR TITLE
`merkledb` / `sync` -- remove TODOs

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

I think these TODOs are TODONT's

## How this works

Remove comments

## How this was tested

N/A

### CODE REVIEW

File: x/merkledb/db.go [hunk 1]
```diff
@@ -566,7 +566,6 @@ func (db *merkleDB) GetChangeProof(
 		return i.Compare(j) < 0
 	})
 
-	// TODO: sync.pool these buffers
```
Comment (danlaine) [inline x/merkledb/db.go:569]: I don't think this would be worth the added cognitive load

File: x/sync/manager.go [hunk 2]
```diff
@@ -190,10 +189,6 @@ func (m *Manager) sync(ctx context.Context) {
 		default:
 			m.processingWorkItems++
 			work := m.unprocessedWork.GetWork()
-			// TODO danlaine: We won't release [m.workLock] until
-			// we've started a goroutine for each available work item.
-			// We can't apply proofs we receive until we release [m.workLock].
-			// Is this OK? Is it possible we end up with too many goroutines?
```
Comment (danlaine) [inline x/sync/manager.go:196]: This was fixed a while back by the above case:

```go
case m.processingWorkItems >= m.config.SimultaneousWorkLimit:
			// We're already processing the maximum number of work items.
			// Wait until one of them finishes.
			m.unprocessedWorkCond.Wait()
```

File: x/sync/network_client.go [hunk 3]
```diff
@@ -312,7 +312,6 @@ func (c *networkClient) Shutdown() {
 	defer c.lock.Unlock()
 
 	// reset peers
-	// TODO danlaine: should we call [Disconnected] on each peer?
```
Comment (danlaine) [inline x/sync/network_client.go:315]: I don't think this is necessary

Comment (StephenButtolph) [reply x/sync/network_client.go:315]: q: why does this function exist at all? It doesn't seem to cancel any outstanding requests or cleanup any state

**Review by StephenButtolph — APPROVED** [review]

**Comment by danlaine** [comment]
@StephenButtolph 

> q: why does this function exist at all? It doesn't seem to cancel any outstanding requests or cleanup any state

I'm down to remove it

File: x/sync/network_client.go (hunk 3 above)
Comment (danlaine) [reply x/sync/network_client.go:315]: removed

**Review by darioush — APPROVED** [review]

**Pull request merged**